flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -993,9 +1000,34 @@ object JobManager {
       configuration.setString(ConfigConstants.FLINK_BASE_DIR_PATH_KEY, configDir + "/..")
     }
 
-    val hostname = configuration.getString(ConfigConstants.JOB_MANAGER_IPC_ADDRESS_KEY, null)
-    val port = configuration.getInteger(ConfigConstants.JOB_MANAGER_IPC_PORT_KEY,
-      ConfigConstants.DEFAULT_JOB_MANAGER_IPC_PORT)
+    // HA mode
+    val (hostname, port) = if (ZooKeeperUtil.isJobManagerHighAvailabilityEnabled(configuration)) {
+      // TODO @removeme @tillrohrmann This is the place where the host and random port for JM is
+      // chosen.  For the FlinkMiniCluster you have to choose it on your own.
+      LOG.info("HA mode.")
+
+      if (config.getHost == null) {
```

Comment:
Hmm you're right that this would shift the problem from knowing in which language the object was implemented to whether the method is pure or not. This would require an even deeper knowledge of the existing code. I guess that your proposal to always use parenthesis for Java methods is then the best solution.